src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -272,106 +273,158 @@ private void emitLiteral(Object o) throws IOException {
    * names visible due to inheritance.
    */
   String lookupName(ClassName className) {
-    // Different package than current? Just look for an import.
-    if (!className.packageName().equals(packageName)) {
-      if (conflictsWithLocalName(className)) {
-        return className.toString(); // A local name conflicts? Use the fully-qualified name.
+
+    // Create ClassName from current stack
```

Comment:
Just as a heads up, in general, [comments should be sentences][1]. All your comments are missing periods. Seems minor, I know, but the linked article points out some compelling arguments for it. Plus it help reinforce good commenting habit.

 [1]: nedbatchelder.com/blog/201401/comments_should_be_sentences.html